js/spreadsheet.js

```diff
@@ -67,248 +67,109 @@ var spreadsheet = function(options) {
+    var right = [cellPosition[0], cellPosition[1] + 1];
+    var top = [topShift < 0 ? cellPosition[0] : topShift, cellPosition[1]];
+    var bottom = [cellPosition[0] + 1, cellPosition[1]];
 
-    return dataAt;
+    return [left, right, top, bottom];
   }
 
   /**
-   * Method to get a coordinats which we need to select
+   * The function that surfs through the table data for detecting closest cells with data from the selected cell
    * 
-   * @param {Array} startAt - array of the selected coordinats 
-   * @param {Object} moveTo - object that returned from closestData() function
-   * 
-   * @returns {Array} - coordinats that needs to be selected. Example of the returned data: [[startRow, startCol, endRow, endCol]]
+   * @param {array} tableData - an array of table data received from the hot.getData()
+   * @param {array} cellPosition - an array of the cell coordinates
+   * @param {array} processedCells - an array of the cells we have already checked
+   * @returns {array} returns array of the scaned cells example array[rowIndex][cellIndex] = hasData true/false
    */
-  function coordinatsToSelect(startAt, moveTo) {
-    var firstCol, lastCol, firstRow, lastRow, allData;
-
-    // Returns array of the data from the table with handsontable API
-    allData = hot.getData();
-    startAt = startAt[0];
-
-    if (moveTo.left) {
-      // When data located on the left of the selected cell
-      lastCol = startAt[1];
-
-      // Looking for first col in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = lastCol - 1; i >= 0; i--) {
-        if (allData[startAt[0]][i] === null) {
-          firstCol = i;
-          break;
-        }
-      }
-
-      firstCol = firstCol || 0;
-
-      // Looking for the first row in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = startAt[0]; i >= 0; i--) {
-        if (allData[i][firstCol] === null) {
-          firstRow = i;
-          break;
-        }
-      }
-
-      firstRow = firstRow || 0;
-
-      // Looking for the last row in the array of allData
-      // When we got a null value in the cell it means that we reached the range borders
-      for (var i = firstRow; i < allData.length; i++) {
-        if (allData[i][firstCol] === null) {
-          lastRow = i;
-          break;
-        }
-      }
+  function getDataCoordinates(tableData, cellPosition, processedCells) {
+    if (!processedCells) {
+      processedCells = [];
+      processedCells[cellPosition[0]] = [];
+      processedCells[cellPosition[0]][cellPosition[1]] = true;
+    }
 
-      lastRow = _.max([lastRow - 1, 0]);
-    } else if (moveTo.right) {
-      // When data located on the right of the selected cell
-      firstCol = startAt[1];
+    var cellsGroup = getNearestCells(cellPosition);
+    cellsGroup.push(cellPosition);
```

Comment:
Check for all linting errors.

```suggestion

    cellsGroup.push(cellPosition);
```